### PR TITLE
Event: Workshop on Correctness and Reproducibility for Climate and Weather Software

### LINKED ISSUE #1727: Event Notice: Call for Abstracts - Workshop on Correctness and Reproducibility for Climate and Weather Software
* **What:** Call for Abstracts: Workshop on Correctness and Reproducibility for Climate and Weather Software

* **When:** November 9-10, 2023 

* **Where:** Boulder, Colorado and Online

* **Website:** [*URL for event web site*](https://ncar.github.io/correctness-workshop/)

* **Summary:** This workshop aims to provide a venue to discuss challenges, opportunities, and recent advances in ensuring software correctness and reproducibility for climate and weather modelers, HPC community members, and industry partners. 

* **Description:** 

Abstracts are being accepted now through August 1 for the Workshop on Correctness and Reproducibility for Climate and Weather Software.  This workshop will take place November 9 & 10 at the NCAR’s Mesa Lab and virtually.

**Scope:**

We are soliciting talks related (but not limited to) the following topics:

- Tools and approaches for software testing, debugging, quality assurance, and continuous integration.
- Statistical and ensemble-based approaches for evaluating model consistency and software correctness.
- Software design approaches and development practices for streamlining correctness and reproducibility efforts.
- Formal methods, abstraction, and logical proof techniques for rigorous verification.
- Verifying and validating large-scale applications running on HPC clusters, cloud computing systems, heterogeneous systems, GPUs, etc.
- Other software correctness and reproducibility approaches for facilitating verification and validation.

**Co-chairs:**
- Allison Baker,  National Center for Atmospheric Research
- Alper Altuntas,  National Center for Atmospheric Research

**Committee:**
- Ilene Carpenter, Hewlett Packard Enterprise
- Brian Dobbins, National Center for Atmospheric Research
- Michael Duda, National Center for Atmospheric Research
- Dorit Hammerling, Colorado School of Mines
- Thomas Hauser, National Center for Atmospheric Research
- Karsten Peters-von Gehlen, Deutsches Klimarechenzentrum GmbH (DKRZ)

**Keynote Speakers:**
- Peter Dueben, European Centre for Medium Range Weather Forecasts (ECMWF)
- Steve Easterbrook, University of Toronto
- Dorit Hammerling, Colorado School of Mines
- John Baugh, North Carolina State University

### PR DESCRIPTION
**Pull request by markcmiller86**
EB Member: @bartlettroscoe   

Resolves #1727 

## PR checklist for files displayed on bssw.io site

* [x] ***[Author]*** `@mention` the BSSw.io editorial board member `@<eb-member-id>` in **Description** above assigned to shepherd your PR.
* [x] ***[Author]*** Add the `<issue-id>` in the **Description** above for the associated GitHub Issue.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py])~~.
* [x] ***[Author]*** Inspect the content in the `*.md` file(s) as rendered in GitHub for this PR.
* [x] ***[EB Mem]*** Assign this PR to the EB member `<eb-member-id>`.
* [x] ***[EB Mem]*** Assign this PR to the author of the PR `<pr-author-id>`.
* [x] ***[EB Mem]*** Add one or more Reviewers.
* [x] ***[EB Mem]*** Add label `content: <content-type>` for the type of contribution.
* [x] ***[EB Mem]*** Add to Project `Content Development` (see [Content Development]).
* [x] ***[EB Mem]*** Add [meta-data] to the `*.md` file(s) (set `Publish: yes`).
* [x] ***[EB Mem]*** Add label `preview` (so PR branch will be merged to 'preview' branch and watch for possible merge failures).
* [x] ***[EB Mem]*** Rebuild [preview] site and confirm new content is there, renders correctly and is returned in searches.
* [x] ***[Author]*** Make any final changes to the PR based on feedback.
* ~~[ ] ***[Author]*** Ensure `wikize_refs.py -i <base>.md` is run and commit (if using [wikize_refs.py]).~~
* [x] ***[EB Mem]*** Rebuild [preview] site and re-confirm content looks correct.
* [x] ***[EB Mem]*** Ensure at least one reviewer signs off on the final changes.
* [x] ***[EB Mem]*** Move the PR to "Ready to Publish" in [Content Development].
* [x] ***[EB Mem]*** Leave comment and @mention `@betterscientificsoftware/bssw-maint` (**BSSw Maint**) asking to carry out final publication steps.
* [ ] ***[BSSw Maint]*** Verify that all needed files are present in the PR (article, images, updates to Site/Homepage.md carousel and/or Site/Announcements/Announcements.md as appropriate).
* [ ] ***[BSSw Maint]*** Merge PR. (Should automatically move to "Done" in [Content Development].)
* [ ] ***[BSSw Maint]*** Verify new contribution shows up on [bssw.io] as expected.

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

File: Events/2023-repro-climate.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Workshop on Correctness and Reproducibility for Climate and Weather Software
```
Comment (bartlettroscoe) [inline Events/2023-repro-climate.md:1]: Should we put the year in the title, which will go into the page name on bssw.io?  That way, if there is a future workshop they will not conflict.

Comment (bartlettroscoe) [inline Events/2023-repro-climate.md:1]:
```suggestion
# Workshop on Correctness and Reproducibility for Climate and Weather Software 2023
```

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
I would just add the year to the title as suggested below.  Then I think this is good to go (give this was just copy and pasted by the author).

File: Events/2023-repro-climate.md (hunk 1 above)
Comment (markcmiller86) [reply Events/2023-repro-climate.md]: Good idea. Done.

**Comment by markcmiller86** [comment]
https://github.com/orgs/betterscientificsoftware/teams/bssw-maint please complete remaining steps

**Comment by bartlettroscoe** [comment]
@alperaltuntas, how do you think this looks:

   https://github.com/betterscientificsoftware/bssw.io/blob/mcm86-28jun23-add-repro-climate-event/Events/2023-repro-climate.md

?

**Comment by alperaltuntas** [comment]
Looks great. Thanks!

**Review by bartlettroscoe — APPROVED** [review]
Looks good.  Thanks!

**Comment by rinkug** [comment]
@markcmiller86 : The event file was missing the event logistics information in the deck; therefore it was showing up in past events. Its rectified now and preview shows well. I am going to go ahead and merge it

**Comment by rinkug** [comment]
Published version: https://bssw.io/events/workshop-on-correctness-and-reproducibility-for-climate-and-weather-software-2023

Thank you for contributing @alperaltuntas and thanks for shepherding @markcmiller86

**Comment by markcmiller86** [comment]
> The event file was missing the event logistics information in the deck; therefore it was showing up in past events. 

Ah. Ok. Well, silly me. I deleted what appeared to me to be duplicate information. That in formation is listed twice. Once as a series of mardown bullets with keyword leading words which apparently *only* the backend consumes and once again as a table that only the front end displays.

Looking at the raw markdown from another example, `2023-usrse-conf.md`...

```
- Deadline for workshops, tutorials, and BoFs: April 4, 2023
- Deadline for papers and notebooks: May 1, 2023
- Deadline for poster and talk abstracts: June 19, 2023
- Event Dates: October 16-18, 2023
- Location: Chicago, Illinois
- Website: https://us-rse.org/usrse23/
- Organizers: US-RSE Association

#### Contributed by [Miranda Mundt](https://github.com/mrmundt/)

#### Publication date: March 13, 2023

Event Information | Details
:--- | :---
Event Name | [United States Research Software Engineer Association Conference 2023 (US-RSE'23)](https://us-rse.org/usrse23/)
Deadline for workshops, tutorials, and BoFs| April 4, 2023
Deadline for papers and notebooks| May 1, 2023
Deadline for poster and talk abstracts| June 19, 2023
Event Dates| October 16-18, 2023
Website | https://us-rse.org/usrse23/
```

It is not clear to a content creater why there is duplicate content. It might help to preface the bulleted list (which apparently gets removed by the back end) with a comment `<!-- This block is actually metadata, not markdown -->`. Or, better yet, to make it clearer just *embed* it in an XML comment so a would-be content creator can tell its purpose.

A final thought is, why can't these two things be combined into a single block of content which serves both purposes?

**Comment by bernhold** [comment]
@markcmiller86 you've figured it out: the list is, in effect, metadata, while the table is content that we've decided we want to display in the body.

The multi-date capability in the front-end looks for entries in that list that have "date" or "deadline" in them and, effectively, sorts them in chronological data.  The front-end always displays exactly one of the dates from that list, whichever is chronologically next.  So in the US-RSE case, it would have shown the workshop deadline up to 4 April, then the paper deadline up to 1 May, etc.  Now it is showing the event date.

The comment idea is reasonable, but we need to verify that it will actually work.  I've had problems with HTML (XML) comments around that metadata list.  In my case, the comment block wasn't being respected and dates within the comment block were being interpreted as part of the metadata.  I've asked for HTML comments to be respected in that part of the document, but I don't think it has been implemented yet.

As to why the duplication: the whole table in the body is a duplication of information in the deck or elsewhere in the article.  But some time ago, that's what we decided we wanted the format to be.  As I recall, the idea was to summarize in one place.  In theory, I guess we could generate that table, but that would require changing the front-end, so, as we currently do things, we have to ask Parallactic to do it.

**Pull request merged**